addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLCustomParser.java

```diff
@@ -79,6 +80,7 @@
     private SOAPMsgConfig lastConfig; // Only used for unit testing purposes.
     private final TableWsdl table;
     private ValueGenerator valueGenerator;
+    private boolean syncParse = true;
```

Comment:
This should be up to the caller not a "global" state (the same instance is called by all functionality), from the API/Spider it should be always synchronous and only asynchronous when importing using the UI.
That's an existing issue so fine if not fixed in this PR but it should be noted (`FIXME` or `TODO`) that this is temporary.